CONTRIBUTING.md

```diff
@@ -40,8 +40,8 @@ A *config* knows about the code style rules and the files and directories that m
 You can do some things to increase the chance that your pull request is accepted without communication ping-pong between you and the reviewers:
 
 * Submit [single](https://en.wikipedia.org/wiki/Single-responsibility_principle) pull request per fix or feature.
-* Don't amend commits because it makes review rounds harder - all commits from your branch will be squashed (without commit messages) during merge, so you can treat pull request as a playground, without keeping everything tidy at any point.
-* If your changes are not up-to-date, [rebase](https://git-scm.com/docs/git-rebase) your branch onto the parent branch. Do it regularly whenever your branch is behind `master` branch, that will eliminate risk of problems after the merge.
+* Keep meaningful commit logs, don't use meaningless messages (e.g. `foo`, `more work`, `more work`, `more work`) and don't push complex PR as a single commit.
+* Don't override your branch history with [rebase](https://git-scm.com/docs/git-rebase) or [commit --amend](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend), because it makes review rounds harder and also complicates work for PRs built on top of your PR. All commits from your branch will be squashed (without commit messages) during the merge.
```

Comment:
👎

I prefer rebase over syncing with a merge. If you make stacked PR (branch on top of branch) it's easy to rebase part of history with `--onto`.

Reply:
:-1: I prefer syncing over rebase and later attempt to find which comment was rebased and which got added on top as new content. Got multiple cases over time when "mark as read" was not working properly after rebase, while always after sync with master.

Reply:
I totally don't agree but it's not a discussion for PR. Anyway, when it comes to "Viewed" chechbox I believe it's not about rebase vs merge but rather push after sync vs push after sync + new commits. Both approaches will reset review state when the file was changed after sync.

Reply:
agree to disagree.
Let's go with middleground then - you prefer one way, I another, there was given way enforced because of imperfection of tool we were using to do PR merges (it blew up when PR had a merge...). Now, `merge` button can easily handle both variants - let's not enforce one approach on user if not needed and we have no aligned preference.

```suggestion
* Don't override your branch history with [rebase](https://git-scm.com/docs/git-rebase) or [commit --amend](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend), because it makes review rounds harder and also complicates work for PRs built on top of your PR. All commits from your branch will be squashed (without commit messages) during the merge.
* * Don't [amend](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend) commits because it makes review rounds harder - all commits from your branch will be squashed (without commit messages) during the merge.
```

Reply:
agree to disagree.

Let's go with middleground then - you prefer one way, I another, there was given way enforced because of imperfection of tool we were using to do PR merges (it blew up when PR had a merge...). Now, `merge` button can easily handle both variants - let's not enforce one approach on user if not needed and we have no aligned preference.

```suggestion
* Don't [amend](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend) commits because it makes review rounds harder - all commits from your branch will be squashed (without commit messages) during the merge.
```

Reply:
agree to disagree.

Let's go with middleground then - you prefer one way, I another, there was given way enforced because of imperfection of tool we were using to do PR merges (it blew up when PR had a merge...). Now, `merge` button can easily handle both variants - let's not enforce one approach on user if not needed and we have no aligned preference.

```suggestion
* Don't [amend](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend) commits because it makes review rounds harder - all commits from your branch will be squashed (without commit messages) during the merge.
```

Reply:
```suggestion
* Don't [amend](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---amend) commits because it makes review rounds harder - all commits from your branch will be squashed (without commit messages) during the merge.
```